Add `llvm-skip-rebuild` flag to `x.py`

matthew-healy opened the pull request:
This PR follows on from #67437 to complete the feature request from #65612.

Specifically it adds a new command-line flag, `--llvm-skip-rebuild`, which overrides both any value set in `config.toml` and the default value (`false`).

I'm not 100% confident that I've implemented the override in the "best" way, but I've checked it locally and it seems to work at least.

This option isn't currently mentioned in the Guide to Rustc Development. I'd be happy to write something on it if folk think that's worthwhile.

rust-highfive commented:
r? @alexcrichton

(rust_highfive has picked a reviewer for you, use r? to override)

Centril commented on src/bootstrap/flags.rs:
```diff
@@ -487,6 +497,9 @@ Arguments:
             deny_warnings: parse_deny_warnings(&matches),
+            llvm_skip_rebuild: matches
+                .opt_str("llvm-skip-rebuild")
+                .and_then(|s| s.parse::<bool>().ok()),
```
```suggestion
                .map(|s| s.parse::<bool>().expect("`llvm-skip-rebuild` should be true/false")),
```

Centril commented:
r? @Centril 

r=me with comment ^-- addressed.

matthew-healy commented:
@Centril Thanks for the review! I've updated the PR. I also added a `to_lowercase()` so that users don't have to worry about the casing requirements for correctly parsing a `bool`. Happy to remove that if you'd rather not have it, though.

Centril commented:
It's not something we typically do, but it also seems fine to me. :)

@bors r+

bors commented:
:pushpin: Commit 7e50b599bfeb75f7be1d5a1fa855e37ec6d0e65d has been approved by `Centril`

<!-- @bors r=Centril 7e50b599bfeb75f7be1d5a1fa855e37ec6d0e65d -->
<!-- homu: {"type":"Approved","sha":"7e50b599bfeb75f7be1d5a1fa855e37ec6d0e65d","approver":"Centril"} -->
